pkg/tenant/topics.go

```diff
@@ -367,9 +369,9 @@ func (s *Server) TopicDelete(c *gin.Context) {
 		return
 	}
 
-	// Verify that we have the right token
-	if token.ID.Compare(topic.ID) != 0 {
-		log.Warn().Err(err).Str("token_id", token.ID.String()).Str("topic_id", topic.ID.String()).Msg("confirmation token does not match topic")
+	// Verify that the right token was provided
+	if confirm.Token != topic.ConfirmDeleteToken {
```

Comment:
Note that the token business is specific to the `TopicDelete` endpoint because we want to make it extra hard for a user to delete a topic (since it permanently destroys data).